File: app/models/pet.rb
class Pet < ApplicationRecord
  # Enums
  enum gender: { male: 0, female: 1, unknown: 2 }

  # Associations
  belongs_to :location, optional: true
  has_one :pet_adoption, dependent: :destroy
  has_many :injury_reports, dependent: :destroy
  has_many :medication_schedules, dependent: :destroy
  has_many :vaccination_schedules, dependent: :destroy
  has_many :pet_foods, dependent: :destroy
  has_many :checks_schedules, dependent: :destroy

  # Validations
  validates :name, presence: true
  validates :pet_type, presence: true
  validates :birthday, presence: true
  validates :date_admitted, presence: true
end

File: app/javascript/types/pet.ts
export interface Pet {
  id: number;
  pet_type: string;
  breed: string;
  picture?: string;
  birthday: string;
  name: string;
  nickname?: string;
  gender?: string;
  date_admitted: string;
  notes?: string;
  created_at?: string;
  updated_at?: string;
}

export interface PetFormData {
  pet_type: string;
  breed: string;
  picture?: string;
  birthday: string;
  name: string;
  nickname?: string;
  gender?: string;
  date_admitted: string;
  notes?: string;
}
